src/partition/cuda/partition_op.cu

```diff
@@ -56,6 +56,22 @@ __global__ void _MapLocalIndexByRemainder(
   }
 }
 
+template<typename IdType>
+__global__ void _MapGlobalIndexByRemainder(
+    const IdType * const in,
+    IdType * const out,
+    const int part_id,
+    const int64_t num_items,
+    const int comm_size) {
+  const int64_t idx = threadIdx.x+blockDim.x*blockIdx.x;
+
+  assert(part_id < comm_size);
+
+  if (idx < num_items) {
```

Comment:
Do we need a for loop here?

Reply:
No, this is always launched with enough threads to cover the `num_items`. I've added assertions to all of these kernels to ensure this is the case (when run in debug mode), in 
50a9b26.